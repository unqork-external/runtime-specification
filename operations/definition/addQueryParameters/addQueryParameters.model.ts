import { Const, Description, DiscriminatorValue, Required } from '@tsed/schema'

import { AddQueryParametersOperationOptions } from './addQueryParameters.options'
import { Alpha, DisplayName } from '../../../../decorators'
import { AutogenDisabled } from '../../decorators/autogenDisabled.decorator'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DisplayName('Add Query Parameters')
@DiscriminatorValue(OperationTypes.ADD_QUERY_PARAMETERS)
@Description('Adds new query parameters to the URL.')
@Alpha()
@AutogenDisabled()
export class AddQueryParametersOperation extends Operation<OperationTypes.ADD_QUERY_PARAMETERS> {
  @Const(OperationTypes.ADD_QUERY_PARAMETERS)
  public type = OperationTypes.ADD_QUERY_PARAMETERS as const

  @Required()
  public options: AddQueryParametersOperationOptions = new AddQueryParametersOperationOptions()
}
