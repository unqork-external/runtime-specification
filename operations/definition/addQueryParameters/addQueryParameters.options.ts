import { Default, Description, Optional, Required } from '@tsed/schema'

import { OperationOptions } from '../../interface'

export class AddQueryParametersOperationOptions extends OperationOptions {
  @Description('Parameters to add to the URL')
  @Required()
  params: Record<string, string | number | boolean>

  @Description(
    'Controls behavior when adding a parameter that already exists in the URL. If `true`, the operation will replace the old query parameters with the same key. If `false`, the new query params will be appended.',
  )
  @Default(true)
  @Optional()
  replaceDuplicates?: boolean = true

  @Description(
    'Determines if changing the params should cause the page to redirect to the url with the new query parameters.',
  )
  @Default(false)
  @Optional()
  redirect?: boolean = false

  @Description(
    'Determines the location in which this operation will update query parameters. Either in the default location as specified by RFC 3968 Section 3.4, or appended to in the URL fragment (RFC 3968 Section 3.5).',
  )
  @Default('default')
  @Optional()
  queryParamLocation?: 'default' | 'fragment' = 'default'
}
