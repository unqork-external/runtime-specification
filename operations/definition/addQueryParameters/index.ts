export * from './addQueryParameters.model'
export * from './addQueryParameters.options'
