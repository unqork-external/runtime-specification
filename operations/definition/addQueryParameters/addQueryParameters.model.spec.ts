import { AddQueryParametersOperation } from './addQueryParameters.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: AddQueryParameters Tests', () => {
  it('AddQueryParameters Model', () => {
    const { schema } = generateSchemaAndValidate(AddQueryParametersOperation)

    expect(schema).toMatchSnapshot()
  })
})
