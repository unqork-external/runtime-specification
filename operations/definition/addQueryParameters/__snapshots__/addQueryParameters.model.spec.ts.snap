// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Operation Model: AddQueryParameters Tests AddQueryParameters Model 1`] = `
{
  "autogenDisabled": true,
  "definitions": {
    "AddQueryParametersOperationOptions": {
      "properties": {
        "params": {
          "description": "Parameters to add to the URL",
          "type": "object",
        },
        "queryParamLocation": {
          "default": "default",
          "description": "Determines the location in which this operation will update query parameters. Either in the default location as specified by RFC 3968 Section 3.4, or appended to in the URL fragment (RFC 3968 Section 3.5).",
          "type": "string",
        },
        "redirect": {
          "default": false,
          "description": "Determines if changing the params should cause the page to redirect to the url with the new query parameters.",
          "type": "boolean",
        },
        "replaceDuplicates": {
          "default": true,
          "description": "Controls behavior when adding a parameter that already exists in the URL. If \`true\`, the operation will replace the old query parameters with the same key. If \`false\`, the new query params will be appended.",
          "type": "boolean",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
      },
      "required": [
        "params",
      ],
      "type": "object",
    },
  },
  "description": "Adds new query parameters to the URL.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/AddQueryParametersOperationOptions",
    },
    "type": {
      "const": "ADD_QUERY_PARAMETERS",
      "examples": [
        "ADD_QUERY_PARAMETERS",
      ],
      "type": "object",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
